google/gax/__init__.py

```diff
@@ -482,3 +489,160 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    if not any_pb.Is(pb_type.DESCRIPTOR):
+        raise TypeError(
+            'Could not convert {} to {}'.format(
+                any_pb.__class__.__name__, pb_type.__name__))
+
+    return pb_type.FromString(any_pb.value)
```

Comment:
I based it off [the existing Operation](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/core/google/cloud/operation.py#L64). It only works with proto types. I assume the result and metadata types will always be protos.